src/Bitcoin/TransactionBuilder.cpp

```diff
@@ -93,6 +100,7 @@ TransactionPlan TransactionBuilder::plan(const Bitcoin::Proto::SigningInput& inp
             // Compute fee.
             // must preliminary set change so that there is a second output
             if (!maxAmount) {
+                assert(input.amount() <= plan.availableAmount);
```

Comment:
Why we use assert here?

Reply:
Fee is computed later as the availAmount - amount; if this assumption does not hold fee would be negative.  This is ensured by safety checks above.